src/App.php

```diff
@@ -1091,8 +1082,8 @@ protected function outputResponse(string $data, array $headers): void
             }
         }
 
-        if ($lateErrorStr !== null) {
-            echo $lateErrorStr;
+        if ($lateError !== null) {
+            echo "\n" . '!! ATK4 UI ERROR: ' . $lateError . ' !!' . "\n";
```

Comment:
I would change "ATK4 UI" to `$app->title` if possible here to enable white labeling of application or just remove it

Reply:
This is critical error that should never apper in production.

Reply:
We already have simillar error in core for exception renderer error.